src/views/ApartmentApp/components/StudentApplication/index.jsx

```diff
@@ -620,7 +620,14 @@ const StudentApplication = ({ userProfile }) => {
         );
         // No additional `else` is needed for this, since `isApplicantValid` handles the `createSnackbar` internally
         if (validApplicants.every((v) => v)) {
-          const saveResult = await housing.saveApartmentApplication(applicationDetails);
+          // Quick fix since the API is trying
+          //  to validate the public profile object but there is no Hall property
+          let applicationDetailsNoProfiles = applicationDetails;
+          applicationDetailsNoProfiles.Applicants = applicationDetails.Applicants.map((a) => {
+            a.Profile.Hall = '';
+            return a;
+          });
```

Comment:
Out of curiosity, wouldn't it be more simple to do something like
```suggestion
          applicationDetailsNoProfiles.Applicants.forEach((a) => a.Profile.Hall = '');
```